### PR TITLE
Move away from git submodules!

### PR DESCRIPTION
**Pull request by t-hiroyoshi**
ref: https://github.com/scala-native/scala-native/issues/61

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 scalalib/src/
 target/
+submodules/scala
 *.dot
+.idea
+.idea_modules
```
Comment (sjrd) [inline .gitignore]: This is environment-specific. It should be in your global `.gitignore`, not in project-specific `.gitignore`s.

File: build.sbt [hunk 2]
```diff
@@ -1,125 +1,11 @@
...
+val nir = Build.nir
+val tools = Build.tools
+val nscplugin = Build.nscplugin
+val sbtplugin = Build.sbtplugin
+val rtlib = Build.rtlib
+val nativelib = Build.nativelib
+val javalib = Build.javalib
+val scalalib = Build.scalalib
+val demoNative = Build.demoNative
+val demoJVM = Build.demoJVM
```
Comment (sjrd) [inline build.sbt]: Why would you move all these things to `.scala`?

File: project/Build.scala [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  val libSettings =
+    baseSettings ++
+      ScalaNativePlugin.projectSettings ++
+      Seq(
+        scalaVersion := libScalaVersion,
+        nativeEmitDependencyGraphPath := Some(file("out.dot"))
+      )
+
+  lazy val util: Project =
+    Project(
```
Comment (sjrd) [inline project/Build.scala]: Don't use the `Project()` constructor. It mostly disappears in sbt 1.0. Especially it doesn't have the `settings` parameter anymore.

File: build.sbt (hunk 2 above)
Comment (t-hiroyoshi) [inline build.sbt]: I think it's better for controlling tests.

Comment (sjrd) [inline build.sbt]: Er ... I don't understand. What do you mean by "controlling tests"?

**Comment by densh** [comment]
The one from @andreaTP is a bit less intrusive, we'll go with that one. Thanks for effort anyway, @t-hiroyoshi !

File: build.sbt (hunk 2 above)
Comment (t-hiroyoshi) [inline build.sbt]: Tests with Build.scala is more readable.
But I find this https://github.com/sbt/sbt/pull/2524.
Using build.sbt is better now.

**Pull request closed without merging**